### DISCUSSION TITLE
Cloudflare Support

### DISCUSSION THREAD

**Discussion by tejaskamath** [supabase/supabase#6377] [Questions]
Hello, first of all let me start by saying really love the work being done @ Supabase. Look forward to every upcoming launch and community day, and have saved a number of hours using Supabase in my projects.

Am currently working on a stack that comprises of Remix + Cloudflare Runtime + Supabase. However realized that XHR does not exist on Cloudflare Runtimes and Supabase uses XHR within their js client which is causing the application to not build.

Is there a way to work around this?

**Comment by inian** [comment]
Hey the client libraries should use the globally available fetch when available and if it doesn't it will fall back to using `cross-fetch` which uses XHR under the hood. 

Can you update your client library to the latest version and see if it works? If it doesn't, a code snippet of what you are trying to run will be helpful Thanks!

**Reply by tejaskamath** [reply]
Hi so have the latest version of supabase js client installed(1.33.3) and have attempted to modify the fetch. Still throws the error:
[pages:err] TypeError: globalThis.XMLHttpRequest is not a constructor

Code usage:
`
export const action: ActionFunction = async ({ request }) => {
  supabase.auth.signIn({ email: "user@gmail.com" });
  return json({
    success: true,
  });
};
`

Supabase definition
`
export const supabase = createClient(SUPABASE_URL, SUPABASE_ANON_KEY, {
  fetch: (...args) => fetch(...args),
});
`

**Reply by kiwicopple** [reply]
Could you try this one please @tejaskamath ? 

https://supabase.com/docs/reference/javascript/initializing#custom-fetch-implementation

It was working previously, so if the instructions above don't work I think we must have some regression

**Reply by tejaskamath** [reply]
@kiwicopple This worked! Thank you so much. Will revert if there is anything that still falters
